### ISSUE TITLE
Logstash service does not start with systemd

### ISSUE THREAD

**Issue by pietervogelaar** [elastic/puppet-logstash#302]
I'm running Logstash on a RedHat 7.2 OS, but Puppet can't start the service.

Also if I run "systemctl start logstash" it won't start. But if I run "service logstash start" it works fine. So at the moment the service script seems not compatible with systemd.

I think this issue is also related: https://github.com/elastic/puppet-logstash/issues/300

**Comment by jarpy** [comment]
Yes, quite a few issues around systemd, unfortunately. See #286 for more gory details.

The short version is that the Logstash 2.x packages do not, themselves, have true systemd support. That makes it hard for Puppet (and this module) to manage the service on systemd distros. Logstash 5 (coming soon) has solid support for systemd, so I expect the situation to improve markedly when it's out.

**Comment by chilcano** [comment]
I'm facing the same issue:

- Logstash: 2.4.0
- CentOS: 7
- Puppet: 4.5.0
 
In my case, `systemctl logstash start|stop|restart|status` doesn't work, and `service logstash start|stop|restart|status` works by setting:

```
class { "logstash":
			    java_install => false,
			    			status => "enabled",
			    service_provider => "init",
			    ...
}
```
Also, the `logstash.RedHat.erb` and `logstash.Debian.erb` didn't work.

Is there any plan to fix that for this version of LS and CentOS by using this Puppet module ?.
Regards.

**Comment by jarpy** [comment]
That's nasty all right. There's no plan to fix this for Logstash 2, because it uses SysV compatibility for init, and because Puppet interacts with that in weird ways.

The plan is to fix all this for Logstash 5.0, which is now generally available and includes fully fledged systemd units.

**Comment by jarpy** [comment]
With the completion of work in #312, this issue is resolved for current Logstash versions.
